lib/core/data/user/user_repository_impl.dart

```diff
@@ -27,20 +31,67 @@ class UserRepositoryImpl implements UserRepository {
   }
 
   @override
-  Future<Either<Failure, User>> getUserProfile({ String? userId, String? username }) async {
+  Future<Either<Failure, User>> getUserProfile({String? userId, String? username}) async {
     assert(userId != null || username != null);
-    final result = await _gqlClient.query(QueryOptions(
-      document: getUserQuery,
-      parserFn: (data) {
-        return User.fromDto(UserDto.fromJson(data['getUser']));
-      },
-      variables: {
-        'id': userId,
-        'username': username
-      }
-    ));
+    final result = await _gqlClient.query(
+      QueryOptions(
+        document: getUserQuery,
+        parserFn: (data) {
+          return User.fromDto(UserDto.fromJson(data['getUser']));
+        },
+        variables: {'id': userId, 'username': username},
+      ),
+    );
 
     if (result.hasException) return Left(Failure());
     return Right(result.parsedData!);
   }
+
+  @override
+  Future<Either<Failure, bool>> updateUserProfile({
+    required String username,
+    String? uploadPhoto,
+    OnboardingGender? gender,
+    String? displayName,
+    String? shortBio,
+  }) async {
+    final result = await _gqlClient.mutate(
+      MutationOptions(
+        document: updateUserProfileQuery,
+        variables: {
+          'username': username,
+          'pronoun': gender?.name.capitalize(),
+          'description': shortBio,
+          'display_name': displayName,
+        }..removeWhere((key, value) => value == null),
+        parserFn: (data) {
+          return data['updateUser'];
+        },
+      ),
+    );
+
+    if (result.hasException) {
+      return Left(Failure());
+    }
+    return Right(result.parsedData != null);
+  }
+
+  @override
+  Future<Either<Failure, bool>> checkValidUsername({required String username}) async {
```

Comment:
apply CheckValidUsernameInput

Reply:
Only one variable here to create entire class is overkill, I suggest we should not do it.